octane/evmengine/keeper/evmmsgs.go

```diff
@@ -16,17 +17,25 @@ import (
 func (k *Keeper) evmEvents(ctx context.Context, blockHash common.Hash) ([]*types.EVMEvent, error) {
 	var events []*types.EVMEvent
 	for _, proc := range k.eventProcs {
-		ll, err := proc.Prepare(ctx, blockHash)
-		if err != nil {
-			return nil, errors.Wrap(err, "prepare msgs")
-		}
-		for _, log := range ll {
-			if err := log.Verify(); err != nil {
-				return nil, errors.Wrap(err, "verify log")
+		err := retryForever(ctx, func(ctx context.Context) (bool, error) {
+			ll, err := proc.Prepare(ctx, blockHash)
+			if err != nil {
+				log.Warn(ctx, "Failed fetching evm logs (will retry)", err, "proc", proc)
+				return false, nil
 			}
-		}
+			// Verify all logs
+			for _, l := range ll {
+				if err := l.Verify(); err != nil {
```

Comment:
It looks like this `err` is not a transient but a deterministic one, so retrying seems to be futile. Or could we have topics without an `err` in line `21` that are not valid when we fetch them first, but are valid on a retry fetch?

Reply:
returning an error doesn't retry. but sure, I'll move it down for clarity

Reply:
I added some more comments to make this clear